rclpy/test/test_node.py

```diff
@@ -800,6 +800,13 @@ def test_use_sim_time(self):
         self.assertTrue(temp_node.get_parameter(USE_SIM_TIME_NAME).value)
         temp_node.destroy_node()
 
+    def test_node_declare_parameters_value_and_type_mismatch(self):
```

Comment:
either way we take, i think it would be worth to keep this test scenario to make sure implementation.